Library/Homebrew/cmd/update-bash.sh

```diff
@@ -2,22 +2,41 @@ brew() {
   "$HOMEBREW_BREW_FILE" "$@"
 }
 
+git() {
+  [[ -n "$HOMEBREW_GIT" ]] || exit 1
+  "$HOMEBREW_GIT" "$@"
+}
+
 which_git() {
-  local which_git
+  local git_path
   local active_developer_dir
 
-  which_git="$(which git 2>/dev/null)"
-  if [[ -n "$which_git" && "/usr/bin/git" = "$which_git" ]]
+  if [[ -n "$HOMEBREW_GIT" && -x "$HOMEBREW_GIT" ]]
+  then
+    git_path="$HOMEBREW_GIT"
+  elif [[ -n "$GIT" && -x "$GIT" ]]
+  then
+    git_path="$GIT"
+  else
+    git_path="$(which git 2>/dev/null)"
+  fi
+
+  if [[ -n "$git_path" ]]
+  then
+    git_path="$(chdir "${git_path%/*}" && pwd -P)/${git_path##*/}"
```

Comment:
It behaves differently if the string doesn't contain a slash at all, but otherwise it's doing the same. For the mentioned case `dirname` will return `.` while `${git_path%/*}` will return the original string.